support-files/mysql.spec.sh

```diff
@@ -124,104 +120,42 @@
 %endif
 
 # ----------------------------------------------------------------------------
-# Packager
+# Distribution support
 # ----------------------------------------------------------------------------
-%if %{undefined mysql_packager}
-%define mysql_packager MySQL Build Team <build@mysql.com>
+
+# Disable post build checks for time being.
+BuildConflicts: post-build-checks rpmlint
+
+BuildRequires: gcc-c++ ncurses-devel perl zlib-devel cmake libaio-devel bison flex
+
+%if 0%{?rhel} == 6 || 0%{?rhel} == 7 || 0%{?fedora} == 20 || 0%{?fedora} == 21
+BuildRequires: time
 %endif
 
-# ----------------------------------------------------------------------------
-# Distribution support
-# ----------------------------------------------------------------------------
-%if %{undefined distro_specific}
-%define distro_specific 0
+%if 0%{?suse_version}
+%if 0%{?suse_version} == 1110
+BuildRequires: gdbm-devel gperf openldap2-client procps pwdutils
 %endif
-%if %{distro_specific}
```

Comment:
Ok, that is a convincing reason. I had not expected this. With some googling, I found this quote:
    But when evaluating BuildRequires, the RPM parser of the
    BuildService runs in an environment where subshell expansions are
    disabled.
https://en.opensuse.org/openSUSE:Specfile_guidelines#OBS_Caveat
Which to me implies that the whole logic of this block will not work in OBS, as it is based on running "test", "rpm" and "sed" commands.

About using "%dist": This seems to be standard (except on RedHat 5 which is obsolete), so IMO it is a good idea to use it.